openmc/settings.py

```diff
@@ -776,16 +786,15 @@ def trace(self, trace: Iterable):
         self._trace = trace
 
     @track.setter
-    def track(self, track: typing.Iterable[int]):
-        cv.check_type('track', track, Iterable, Integral)
-        if len(track) % 3 != 0:
-            msg = f'Unable to set the track to "{track}" since its length is ' \
-                  'not a multiple of 3'
-            raise ValueError(msg)
-        for t in zip(track[::3], track[1::3], track[2::3]):
+    def track(self, track: typing.Iterable[typing.Iterable[int]]):
+        cv.check_type('track', track, Iterable)
+        for t in track:
+            if len(t) != 3:
+                msg = f'Unable to set the track to "{t}" since its length is not 3'
+                raise ValueError(msg)
             cv.check_greater_than('track batch', t[0], 0)
-            cv.check_greater_than('track generation', t[0], 0)
-            cv.check_greater_than('track particle', t[0], 0)
+            cv.check_greater_than('track generation', t[1], 0)
+            cv.check_greater_than('track particle', t[2], 0)
```

Comment:
```suggestion
            cv.check_greater_than('track generation', t[1], 0)
            cv.check_greater_than('track particle', t[2], 0)
            cv.check_type('track generation', t[1], Integral)
            cv.check_type('track particle', t[2], Integral)
```

I don't think this is necessary but I guess the type could be checked here if we want?